### PR TITLE
Change ScrollView to use ResizeObserver

### PR DESCRIPTION
**Pull request by snowystinger**
Better if we can move to ResizeObserver instead of window. We now have a utility function to handle this, though it doesn't do it on layout, so I added that

Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/virtualizer/src/ScrollView.tsx [hunk 1]
```diff
@@ -121,38 +122,33 @@ function ScrollView(props: ScrollViewProps, ref: RefObject<HTMLDivElement>) {
...
 
+  useLayoutEffect(() => {
     updateSize();
-    window.addEventListener('resize', updateSize, false);
-    return () => {
-      window.removeEventListener('resize', updateSize, false);
-    };
-  }, [onVisibleRectChange, ref, state, sizeToFit, contentSize]);
+  }, [updateSize]);
+  useResizeObserver({ref, onResize: updateSize});
```
Comment (LFDanLu) [inline packages/@react-aria/virtualizer/src/ScrollView.tsx:151]: Not 100% sure about this but do we need the `updateSize` here in the dep array? Won't `useResizeObserver` handle calling it if `updateSize` is updated?

Comment (snowystinger) [reply packages/@react-aria/virtualizer/src/ScrollView.tsx:151]: useResizeObserver only attaches the listeners, it doesn't run the function, so we also need to updateSize though anytime visibleRectChange, state, sizeToFit, contentSize, ref change
https://github.com/adobe/react-spectrum/pull/910/files#diff-082f5ada66e52e3e32545d9f58296f93L150

**Review by LFDanLu — APPROVED** [review]
LGTM

File: packages/@react-aria/virtualizer/src/ScrollView.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import React, {
 import {Rect, Size} from '@react-stately/virtualizer';
 import {useLayoutEffect} from '@react-aria/utils';
 import {useLocale} from '@react-aria/i18n';
+import {useResizeObserver} from '@react-spectrum/utils';
```
Comment (devongovett) [inline packages/@react-aria/virtualizer/src/ScrollView.tsx:29]: Probably shouldn't import from react-spectrum in react-aria... do we need to move this util?

**Review by devongovett — APPROVED** [review]

**Pull request merged**